.github/workflows/kind-e2e.yaml

```diff
@@ -0,0 +1,197 @@
+name: KinD e2e tests
+
+on:
+  push:
+    branches: [ 'master' ]
+  pull_request:
+    branches: [ 'master' ]
+
+jobs:
+
+  ko-resolve:
+    name: e2e tests
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false # Keep running if one leg fails.
+      matrix:
+        k8s-version:
+        - 1.17.x
+        - 1.18.x
+        - 1.19.x
+
+        test-suite:
+        - ./test/conformance/runtime
+        - ./test/conformance/api/v1alpha1
+        - ./test/conformance/api/v1beta1
+        - ./test/conformance/api/v1
+        - ./test/e2e
+
+        # Map between K8s and KinD versions.
+        # This is attempting to make it a bit clearer what's being tested.
+        include:
+        - k8s-version: 1.17.x
+          kind-version: v0.7.0
+          # TODO(https://github.com/knative-sandbox/net-istio/issues/285): Make this Istio
+          kingress: contour # istio
+        - k8s-version: 1.18.x
+          kind-version: v0.8.1
+          kingress: kourier
+        - k8s-version: 1.19.x
+          kind-version: v0.9.0
+          kingress: contour
+
+    env:
+      GOPATH: ${{ github.workspace }}
+      GO111MODULE: off
+      KO_DOCKER_REPO: kind.local
+
+    steps:
+    - name: Set up Go 1.14.x
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.14.x
+
+    - name: Install Dependencies
+      run: |
+        GO111MODULE=on go get github.com/google/ko/cmd/ko@master
+
+    - name: Check out code onto GOPATH
+      uses: actions/checkout@v2
+      with:
+        path: ./src/knative.dev/serving
+
+    - name: Install KinD
+      working-directory: ./src/knative.dev/serving
+      run: |
+        set -x
+
+        # Disable swap otherwise memory enforcement doesn't work
+        # See: https://kubernetes.slack.com/archives/CEKK1KTN2/p1600009955324200
+        sudo swapoff -a
+        sudo rm -f /swapfile
+
+        curl -Lo ./kind https://github.com/kubernetes-sigs/kind/releases/download/${{ matrix.kind-version }}/kind-$(uname)-amd64
+        chmod +x ./kind
+        sudo mv kind /usr/local/bin
+
+    - name: Create KinD Cluster
+      working-directory: ./src/knative.dev/serving
+      run: |
+        set -x
+
+        # KinD configuration.
+        cat > kind.yaml <<EOF
+        apiVersion: kind.x-k8s.io/v1alpha4
+        kind: Cluster
+        nodes:
+        - role: control-plane
```

Comment:
I think this gets you a default largest k8s cluster from kind supported at that version rather than changing the version of kind on line 73, you can pick the version of the cluster node image like this: https://github.com/knative-sandbox/eventing-rabbitmq/blob/master/.github/workflows/kind-e2e.yaml#L58

Then we pin to the latest kind version and only change the node image

Reply:
> default largest k8s cluster from kind supported at that version

I care essentially not at all about the node version, we want to test against different versions of the K8s controlplane.  Maybe I'm misunderstanding, we can talk in a bit.

Reply:
my understanding is the kind version is the infra around spinning up the containers and making it all work, but the node versions are the k8s api you are using. I also could be wrong.

Reply:
I asked in `#kind` via https://kubernetes.slack.com/archives/CEKK1KTN2/p1600362869415600 and the short answer is   install the newest verison of kind and select the correct node image for that release.

Reply:
done